bids-validator/bids_validator/rules/file_level_rules.json

```diff
@@ -526,7 +526,7 @@
   },
 
   "microscopy": {
-    "regexp": "^[\\/\\\\](sub-[a-zA-Z0-9]+)[\\/\\\\](?:(ses-[a-zA-Z0-9]+)[\\/\\\\])?microscopy[\\/\\\\](sub-[a-zA-Z0-9]+)(?:(_ses-[a-zA-Z0-9]+))?(?:_sample-[a-zA-Z0-9]+)(?:_acq-[a-zA-Z0-9]+)?(?:_stain-[a-zA-Z0-9]+)?(?:_run-[0-9]+)?(?:_chunk-[a-zA-Z0-9]+)?(((@@@_microscopy_type_@@@)(@@@_microscopy_ext_@@@))|(_photo\\.jpg))$",
+    "regexp": "^[\\/\\\\](sub-[a-zA-Z0-9]+)[\\/\\\\](?:(ses-[a-zA-Z0-9]+)[\\/\\\\])?micr[\\/\\\\](sub-[a-zA-Z0-9]+)(?:(_ses-[a-zA-Z0-9]+))?(?:_sample-[a-zA-Z0-9]+)(?:_acq-[a-zA-Z0-9]+)?(?:_stain-[a-zA-Z0-9]+)?(?:_run-[0-9]+)?(?:_chunk-[a-zA-Z0-9]+)?(((@@@_microscopy_type_@@@)(@@@_microscopy_ext_@@@))|(@@@_photo_ext_@@@))$",
```

Comment:
Is it possible to split the regex in 2 (1 for the `microscopy_ext` and 1 for the `photo_ext`)?

The reason is that microscopy files can have this form:
`sub-<label>[_ses-<label>]_sample-<label>[_acq-<label>][_stain-<label>][_run-<index>][_chunk-<index>]_<suffix>.<microscopy_ext>`
or this form:
`sub-<label>[_ses-<label>]_sample-<label>[_acq-<label>]_photo.<photo_ext>`
where `[]` are for optional `key-value` pairs.

So all the `key-value` pairs are not valid for the `_photo` suffix. The validator should check if the filename correspond to one template or the other.

Note: this does not apply to the `json` extension which is OK as is.

Reply:
@mariehbourget It's definitely possible to split the regex in 2 and I have done this already. To examine the new regex, I renamed the data file under micr_SEM from `sub-01_ses-01_sample-A_SEM.png` to `sub-01_ses-01_sample-A_photo.png` and remained the JSON file `sub-01_ses-01_sample-A_SEM.json` unchanged. With this change, the png file could be recognized correctly by the validator as a microscopy file. However,  an error raised 

`1: [ERR] A json sidecar file was found without a corresponding data file (code: 90 - SIDECAR_WITHOUT_DATAFILE)
                ./sub-01/ses-01/micr/sub-01_ses-01_sample-A_SEM.json
` when I run the validator with `micr_SEM` dataset.

From my perspective, both data and json files followed the naming rules. In this case, this error should not be raised right? Please correct me if I am mistaken.

Reply:
Hi @konantian!

The way I understand the regex now, it works perfectly for all the suffixes that are not `_photo` (for all suffixes listed in `microscopy_type`, ex: `_SEM`).
But with the suffix `_photo`, a name like `sub-01_ses-01_sample-01_acq-01_stain-01_run-01_chunk-01_photo.jpg` would also pass the validation, which is not correct because `stain`, `run` and `chunk` are not allowed with the photo suffix.

For the error in JSON, it is normal because elsewhere in the validator, they check if each json file has a corresponding image file, which was not the case with your example. _photo files don't have json file, so don't worry about that.